x-pack/elastic-agent/pkg/config/config.go

```diff
@@ -47,26 +37,42 @@ func NewConfigFrom(from interface{}, opts ...ucfg.Option) (*Config, error) {
 		opts = DefaultOptions
 	}
 
-	if str, ok := from.(string); ok {
-		c, err := yaml.NewConfig([]byte(str), opts...)
-		return newConfigFrom(c), err
-	}
-
-	if in, ok := from.(io.Reader); ok {
+	var data []byte
+	var err error
+	if bytes, ok := from.([]byte); ok {
+		data = bytes
+	} else if str, ok := from.(string); ok {
+		data = []byte(str)
+	} else if in, ok := from.(io.Reader); ok {
 		if closer, ok := from.(io.Closer); ok {
 			defer closer.Close()
 		}
-
-		content, err := ioutil.ReadAll(in)
+		data, err = ioutil.ReadAll(in)
 		if err != nil {
 			return nil, err
 		}
-		c, err := yaml.NewConfig(content, opts...)
+	} else {
+		c, err := ucfg.NewFrom(from, opts...)
 		return newConfigFrom(c), err
 	}
 
-	c, err := ucfg.NewFrom(from, opts...)
-	return newConfigFrom(c), err
+	var c map[string]interface{}
```

Comment:
can we generalize this a bit? e.g have a list of keywords where we want to skip ucfg resolution. 
i think extraction to func and var list of keys would be enough